.github/workflows/starlingmonkey.yml

```diff
@@ -27,12 +27,88 @@ jobs:
         rustup toolchain install 1.77.1
         rustup target add wasm32-wasi --toolchain 1.77.1
     - name: Build
-      run: npm run build:starlingmonkey:debug
+      run: npm run build:starlingmonkey
     - uses: actions/upload-artifact@v3
       with:
-        name: starling-debug
+        name: starling-release
         path: starling.wasm
 
+  ensure_cargo_installs:
+    name: Ensure that all required "cargo install" commands are run, or we have a cache hit
+    strategy:
+      matrix:
+        include:
+          - crate: viceroy
+            version: 0.9.4 # Note: workflow-level env vars can't be used in matrix definitions
+            options: ""
+          - crate: wasm-tools
+            version: 1.0.28 # Note: workflow-level env vars can't be used in matrix definitions
+            options: ""
+    runs-on: ubuntu-latest
+    steps:
+    - name: Cache ${{ matrix.crate }} ${{ matrix.version }}
+      id: cache-crate
+      uses: actions/cache@v3
+      with:
+        path: "/home/runner/.cargo/bin/${{ matrix.crate }}"
+        key: crate-cache-${{ matrix.crate }}-${{ matrix.version }}
+    - name: Install ${{ matrix.crate }} ${{ matrix.version }}
+      run: cargo install ${{ matrix.crate }} ${{ matrix.options }} --version ${{ matrix.version }}
+      if: steps.cache-crate.outputs.cache-hit != 'true'
+
+  run_wpt:
+    if: github.ref != 'refs/heads/main' && false
```

Comment:
This adds the WPT test suite to CI, but disabled for now as there are still 100 tests failing. Will post fixes in a follow-up.